### PR TITLE
Upgrade istio and pdbs

### PR DESCRIPTION
**Pull request by cliveseldon**
- [x] Upgrade istio to latest 1.16.2
- [x] Upgrade PDB types
- [x] Upgrade k8s api to 1.24
- [x] Update jenkins and github action tests
- [x] Remove deprecated webhook configurations which have reappeared with controller-gen 0.9.2 correct usage

Fixes: #4634 
Fixes: #4539

### CODE REVIEW

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 4c8362dfbc69acda1f6afc9421fbdce0523dd6df | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4645-notebooks-q82gl) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: examples/models/autoscaling/model_with_hpa.yaml [hunk 1]
```diff
@@ -11,7 +11,9 @@ spec:
         metrics:
         - resource:
             name: cpu
-            targetAverageUtilization: 10
+            target:
```
[the remark this answers is not in the document]
Comment (RafalSkolasinski) [reply examples/models/autoscaling/model_with_hpa.yaml:14]: Ok, I see this is the difference from `k8s.io/api/autoscaling/v2beta1` to `k8s.io/api/autoscaling/v2`.

[the remark this answers is not in the document]
Comment (cliveseldon) [reply examples/models/autoscaling/model_with_hpa.yaml:14]: good point- I need to revert that - must have crept in from HPA branch

**Review by RafalSkolasinski — APPROVED** [review]
LGTM. Just would be good to clarify on the CRD changes. 

It seems they are mostly coming from change of `k8s.io/api/autoscaling/v2beta1` to `k8s.io/api/autoscaling/v2`.

Question: How would it behave if  `SeldonDeployment` conforming to old CRD containing
```yaml
        metrics:
        - resource:
            name: cpu
            target:
              type: Utilization
              averageUtilization: 10
```
in its definition be present in the cluster while Core being upgraded to 1.16.0? 

Would it
a) just ignore extra fields ?
b) fail on strict validation?

**Comment by cliveseldon** [comment]
> LGTM. Just would be good to clarify on the CRD changes.
> 
> It seems they are mostly coming from change of `k8s.io/api/autoscaling/v2beta1` to `k8s.io/api/autoscaling/v2`.
> 
> Question: How would it behave if `SeldonDeployment` conforming to old CRD containing
> 
> ```yaml
>         metrics:
>         - resource:
>             name: cpu
>             target:
>               type: Utilization
>               averageUtilization: 10
> ```
> 
> in its definition was present in the cluster while Core being upgraded to 1.16.0?
> 
> Would it a) just ignore extra fields ? b) fail on strict validation?

This is a valid point and maybe requires more consideration as CRDs should be backwards compatible or provide a new version with a hub and spokes method of translation to newer version

**Comment by seldondev** [comment]
@cliveseldon: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 754ff262981b7357bf7e0ed5297842b7f04041e0 | [link](http://dashboard-jx.34.121.152.94.nip.io/namespaces/jx/pipelineruns/io-seldon-core-pr-4645-notebooks-dxr5r) | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by RafalSkolasinski — APPROVED** [review]
LGTM

**Pull request merged**